pkg/api/types.go

```diff
@@ -2766,6 +2767,22 @@ type Preconditions struct {
 	UID *types.UID
 }
 
+// DeletePropagationPolicy decides whether and how garbage collection will be performed.
+type DeletePropagationPolicy string
+
+const (
+	// The default depends on the existing finalizers on the object and the type of the object.
+	DeletePropagationDefault DeletePropagationPolicy = "DeletePropagationDefault"
+	// Orphans the dependents.
+	DeletePropagationOrphan DeletePropagationPolicy = "DeletePropagationOrphan"
+	// Deletes the object from the key-value store, the garbage collector will delete the dependents in the background.
+	DeletePropagationBackground DeletePropagationPolicy = "DeletePropagationBackground"
+	// The object exists in the key-value store until the garbage collector deletes all the dependents whose ownerReference.blockOwnerDeletion=true from the key-value store.
```

Comment:
I would prefer the string constants to be shorter - "Orphan" and "Background".  I don't want to type Propagation in 100 examples.